File: lib/src/core/models/review.dart
class Review {
  const Review({
    required this.id,
    required this.rating,
    required this.comment,
    required this.reviewerName,
  });

  final String id;
  final num rating;
  final String comment;
  final String reviewerName;

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: (json['_id'] ?? json['id'] ?? '') as String,
      rating: (json['rating'] ?? 0) as num,
      comment: (json['comment'] ?? '') as String,
      reviewerName: (json['reviewerName'] ?? '') as String,
    );
  }
}

File: backend/models/Review.js
// models/Review.js
const mongoose = require("mongoose");

const ReviewSchema = new mongoose.Schema(
  {
    teacherId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true
    },
    studentId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      required: true
    },
    matchId: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Match"
    },

    rating: {
      type: Number,
      min: 1,
      max: 5,
      required: true
    },

    comment: {
      type: String,
      default: ""
    }
  },
  { timestamps: true }
);

// Each student can only review a teacher once (simple rule)
ReviewSchema.index({ teacherId: 1, studentId: 1 }, { unique: true });

module.exports = mongoose.model("Review", ReviewSchema);
